### DISCUSSION TITLE
.NET 9+ Package Publishing

### DISCUSSION THREAD

**Discussion by richlander** [dotnet/core#9556] [General]
For .NET 9 and future releases, we will be publishing packages at `packages.microsoft.com` for Azure Linux, Debian,  OpenSUSE, and SLES. We previously published packages for other distros, like Ubuntu. We have stopped publishing packages for distros that publish their own .NET packages.

The reasoning:

- Users with .NET distro-archive packages available are typically best served by using those packages.
- Using packages that are available from multiple feeds can be confusing.

More context: https://github.com/dotnet/core/discussions/9258

Note: distro-archive packages are typically limited to the [.1xx SDK band](https://learn.microsoft.com/dotnet/core/releases-and-support#feature-bands-sdk-only), while `packages.microsoft.com` packages are not. Users that want to use the latest feature band builds (on distros without `packages.microsoft.com` packages available)  must [install those builds manually](https://learn.microsoft.com/dotnet/core/install/linux).

## Packages

We will publish packages for the following distros for .NET 9.

- Azure Linux 3.0
- Debian 12
- [OpenSUSE 15](https://en.opensuse.org/DotNet)
- SLES

We will not publish packages for the following distros for .NET 9.

- Alpine Linux
- CentOS Stream
- [Fedora](https://packages.fedoraproject.org/search?query=dotnet-sdk)
- Red Hat Enterprise Linux
- Mariner (Azure Linux 2.0)
- Ubuntu

See [Supported OS versions](https://github.com/dotnet/core/blob/main/release-notes/9.0/supported-os.md) to see the complete set of supported operating systems.

**Comment by tmds** [comment]
@richlander can you include Alpine and Red Hat Enterprise Linux in the overview?

cc @omajid @ayakael

**Reply by richlander** [reply]
Good call. Updated. Thanks for that.

**Comment by adegeo** [comment]
I think you meant OpenSUSE 15, not 42.3. Correct?

**Reply by richlander** [reply]
Yes. Thanks!

**Comment by antimetahh** [comment]


**Reply by antimetahh** [reply]
https://github.com/dotnet/core/discussions/9556

**Comment by mssucksalot** [comment]
https://www.rpmfind.net/linux/rpm2html/search.php?query=dotnet-sdk-10&submit=Search+...

Currently, .NET 10 is not available on RHEL-based distros. It is only available on Stream-based distros. As a result, applications crash without the SDK because certain methods are not implemented, even when targeting versions 8.0 or 9.0.

It's a strange decision... I can't even imagine something like MariaDB being unavailable on certain distros, even if it is packaged by the OS vendor.

**Reply by tmds** [reply]
We published rpms last week. How are you trying you install it? 

cc @omajid

**Reply by mssucksalot** [reply]
I don’t understand what you mean.

.NET 10 is not available for RHEL-based distros yet (because it is only provided by vendor). If you list all dotnet-sdk-*, you won’t see version 10 available.
There's a significant gap between OpenSUSE, which has direct support from Microsoft, and distributions like Rocky, which rely on RHEL releases.
It would be great to have native Microsoft packages available for all platforms, even if vendor support is already declared.

**Reply by tmds** [reply]
I meant that on RHEL the .NET 10 rpms are available.

The RHEL-based distos follow RHEL packages. It's best to check with them on the status of .NET 10.

**Reply by mssucksalot** [reply]
RHEL - yes, but Oracle - no, Rocky - no, and so on. AlmaLinux already has it available as well because it's based on Stream.

This is not a single issue. NET 8 have different SDK versions like 8.0.416 and 8.0.319.
And generally vendors use not the same one as Microsoft provides in their repos for Linux (Debian or OpenSUSE).
So something that works on the MS build might fail on the vendor’s version.

In other words, what works for you for example in Windows or OpenSUSE might not work in production under RHEL-based systems, Alpine Linux, etc. It doesn’t happen often, but when it does, it’s a real pain. So packages from MS are very appreciated for all platforms.

**Reply by tmds** [reply]
Have you reached out to the maintainers of the distro you use and made them aware of the issue?

**Reply by mssucksalot** [reply]
I am not single who has the same issues.

Usually you receive something like that: "Closing as we only support what is shipped by us and upstream (RHEL)."

Or for example this thread: https://forums.rockylinux.org/t/request-for-updated-net-sdk-packages-in-appstream-for-rocky-linux-9/18388

They wrote:

> 
> The issue here is that versions of these packages in the Rocky repos don’t diverge from what’s in the upstream Red Hat packages. This is how we maintain our compatibility with RHEL. Currently, Red Hat only packages the N.0.1xx versions of the SDK, so until that changes, that’s what Rocky is going to package.
> 
> FWIW, it looks like Microsoft has a repo that would work with Rocky with the N.0.2xx or N.0.4xx SDKs at [Index of /rhel/9/prod/](https://packages.microsoft.com/rhel/9/prod) (disclaimer: I haven’t tested this repo).
> 

But I tested that repo, and dotnet there is not functional.

----

I am personally using OpenSUSE, so I do not have such issues. But I am very limited in my attempts to help others.

**Reply by tmds** [reply]
Users of the distro should reach out to its maintainers to make them aware .NET 10 is available in RHEL but not yet in the downstream distro.

**Reply by mssucksalot** [reply]
This does not eliminate another problem, which is why native packages from MS are preferable.

**Comment by MV10** [comment]
This probably needs to be updated for .NET 10. 
It is linked from the top-level Linux [install page](https://learn.microsoft.com/en-us/dotnet/core/install/linux)

**Reply by richlander** [reply]
I updated the title and description to say ".NET 9+".
